backend/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+use number::FieldElement;
+use pil_analyzer::Analyzed;
+use strum::{Display, EnumString, EnumVariantNames};
+
+#[derive(Clone, EnumString, EnumVariantNames, Display)]
+pub enum Backend {
+    #[strum(serialize = "halo2")]
+    Halo2,
+    #[strum(serialize = "halo2-mock")]
+    Halo2Mock,
+}
+
+/// Create a proof for a given PIL, fixed column values and witness column values
+/// using the chosen backend.
+pub fn prove<T: FieldElement>(
+    pil: &Analyzed<T>,
+    fixed: Vec<(&str, Vec<T>)>,
+    witness: Vec<(&str, Vec<T>)>,
+    backend: Backend,
+    params: Option<Vec<u8>>,
+) -> Option<Vec<u8>> {
+    match backend {
+        Backend::Halo2 => Some(halo2::prove_ast(pil, fixed, witness, params)),
+        Backend::Halo2Mock => {
+            halo2::mock_prove(pil, fixed, witness);
+            None
+        }
+    }
+}
+
+pub fn setup<T: FieldElement>(size: usize, backend: Backend) -> Vec<u8> {
+    match backend {
+        Backend::Halo2 | Backend::Halo2Mock => halo2::generate_params::<T>(size),
```

Comment:
Technically the mock prover does not need parameters. Maybe write an empty vector for that and in `prove` assert that the params are empty?